src/main/resources/com/rultor/agents/req/merge.sh

```diff
@@ -20,6 +20,12 @@ while [ $(git show-branch "${BRANCH}" 2>/dev/null | wc -l) -gt 0 ]; do
 done
 
 git checkout -B "${BRANCH}" "fork/${fork_branch}"
+git fetch origin
+if git log --name-status --oneline HEAD..origin/${head_branch} | grep ".rultor.yml"; then
+  echo "Since original repository has different version of .rultor.yml file you have to sync your fork with the upstream repository."
+  exit 1
+fi
```

Comment:
@original-brownbear Great thanks for advices. I'm really appreciate your help.

> some functionality to fetch the merged profile.

I added puzzle.

> some functionality to verify it does not contain changes to authentication things

I implemented method and covered it by unit tests.

> adjust the actual build to run on the merged profile

We will have this out-of-the box here:  https://github.com/yegor256/rultor/pull/1064/files#diff-eb31f19e43e00bbef161a0fbd2a228a0R152